ci_tools/bot_tools/github_api_interactions.py

```diff
@@ -799,6 +814,67 @@ def request_reviewers(self, pr_id, request_team = False, reviewers = ()):
 
         self._post_request("POST", url, review_requests)
 
+    def wait_for_runs(self, commit_sha, run_names, max_time=15*60, wait_time=15):
+        """
+        Wait for the specified workflow runs associated with the commit.
+
+        Use the API to find workflow runs associated with a commit as described here:
+        https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#list-workflow-runs-for-a-repository
+
+        If the workflows are not finished then wait before trying again. Keep trying until the
+        timeout is reached.
+
+        Parameters
+        ----------
+        commit_sha : str
+            The SHA of the commit for which the workflows are run.
+
+        run_names : tuple[str,...]
+            The names of the workflows which should finish before this method terminates.
+
+        max_time : int, default=15*60
+            The maximum number of seconds which should be spent in this method.
+
+        wait_time : int, default=15
+            The time in seconds that the method should wait for before rechecking the workflow
+            results.
+
+        Returns
+        -------
+        bool
+            True if all workflows succeeded, False otherwise.
+        """
+        url = f"https://api.github.com/repos/{self._org}/{self._repo}/actions/runs"
+        completed = False
+        # Run for 15 mins maximum
+        timeout = time.time() + max_time
+
+        request_result = self._post_request("GET", url, params={'head_sha': commit_sha}).json()
+        workflow_runs = [j for j in request_result['workflow_runs'] if j['name'] in run_names]
+        completed = all(j['status'] == 'completed' for j in workflow_runs)
+
+        print(commit_sha)
+        print(workflow_runs)
+        print([j['name'] for j in workflow_runs])
+        print([j['status'] for j in workflow_runs])
+        print([j['conclusion'] for j in workflow_runs])
```

Comment:
Do we need the prints?

Reply:
Technically no, they were added for debugging purposes. But without them if there is a problem with this code we will have no clues as to what may be causing it. And as it only runs on the `main` branch it will be hard to put them back later. In theory we can test on a fork but if there is a repo-specific problem it will be impossible to debug without these prints.

Given that we will never open the job where these will be printed unless there is a problem I don't see any harm in leaving them